it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnerPartnershipRegisteredAddressControllerISpec.scala

```diff
@@ -0,0 +1,122 @@
+            }
+          }
+        }
+      }
+    }
+  }
+
+  "POST /form/business-partners/partner-address" when {
+    "the form has no errors" should {
+      "return 200" in {
+        given
+          .commonPrecondition
+
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl$requestURL")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).withHttpHeaders(xSessionId,
+            "Csrf-Token" -> "nocheck")
+            .post(Map("partnerAddressLine" -> Seq("1"),
+              "partnerPostcode" -> Seq("AB1 2YZ")))
+
+          whenReady(result) { res =>
+            res.status mustBe 200
+            res.body mustBe "Next page! with postcode: AB1 2YZ and address line 1"
+          }
+        }
+      }
+    }
+
+    "address line not populated" should {
+      "return 200" in {
+        given
+          .commonPrecondition
+
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl$requestURL")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).withHttpHeaders(xSessionId,
+            "Csrf-Token" -> "nocheck")
+            .post(Map("partnerAddressLine" -> Seq.empty,
+              "partnerPostcode" -> Seq("AB1 2YZ")))
+
+          whenReady(result) { res =>
+            res.status mustBe 200
+            res.body mustBe "Next page! with postcode: AB1 2YZ and no address line"
+          }
+        }
+      }
+    }
+
+    "postcode not populated" should {
+      "return 400" in {
+        given
+          .commonPrecondition
+
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl$requestURL")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).withHttpHeaders(xSessionId,
+            "Csrf-Token" -> "nocheck")
+            .post(Map("partnerAddressLine" -> Seq("1"),
+              "partnerPostcode" -> Seq.empty))
+
+          whenReady(result) { res =>
+            res.status mustBe 400
+            val page = Jsoup.parse(res.body)
+            page.getElementsByClass("govuk-error-summary").text() must include("Enter the postcode of the address")
+          }
+        }
+      }
+    }
+
+    "postcode invalid format" should {
+      "return 400" in {
+        given
+          .commonPrecondition
+
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl$requestURL")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).withHttpHeaders(xSessionId,
+            "Csrf-Token" -> "nocheck")
+            .post(Map("partnerAddressLine" -> Seq("1"),
+              "partnerPostcode" -> Seq("A")))
+
+          whenReady(result) { res =>
+            res.status mustBe 400
+            val page = Jsoup.parse(res.body)
+            page.getElementsByClass("govuk-error-summary").text() must include("There is a problem Enter a valid postcode")
+          }
+        }
+      }
+    }
```

Comment:
Add test for when addressLine is longer than 35 characters or also invalid format (including unsupported characters [:, ;...])